File: order-service/src/main/java/me/ketlas/orderservice/entities/Order.java
package me.ketlas.orderservice.entities;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import me.ketlas.orderservice.enums.OrderStatus;
import me.ketlas.orderservice.model.Customer;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "order_tbl")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Temporal(TemporalType.TIMESTAMP)
    private Date createdAt;
    @Enumerated(EnumType.STRING)
    private OrderStatus status;
    private Long customerId;
    @Transient
    private Customer customer;
    @OneToMany(mappedBy = "order")
    private List<ProductItem> productItems;

}

File: web-app/src/app/modules/order.ts
import {Page} from "./page";
import {ProductItem} from "./product-item";

export interface Order {

  id:number,
  createdAt:number,
  status:string,
  customerId:number,
  customer:number,
  productItems:ProductItem[];
}

export interface Embedded{
  orders:Order[];

}
export interface OrderResponse{
  _embedded:Embedded;
  page:Page;
}

